subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepository.java

```diff
@@ -201,11 +201,11 @@ private void snapshotAfterOutputsWereGenerated(TaskHistoryRepository.History his
         private TaskUpToDateState getStates() {
             if (states == null) {
                 HistoricalTaskExecution previousExecution = history.getPreviousExecution();
+                CurrentTaskExecution currentExecution = history.getCurrentExecution();
                 if (previousExecution == null) {
                     states = NoHistoryTaskUpToDateState.INSTANCE;
                 } else {
                     // Calculate initial state - note this is potentially expensive
-                    CurrentTaskExecution currentExecution = history.getCurrentExecution();
```

Comment:
Please move the comment, too. We should also mention that resolving current execution is a requirement in a comment.

Reply:
Moved.